### ISSUE TITLE
FileNotFoundException for project sln file

### ISSUE THREAD

**Issue by Dante210** [JetBrains/resharper-unity#425]
System.IO.FileNotFoundException: Could not find file "C:\work\repos\a\a.sln".
File name: 'C:\work\repos\a\a.sln'
  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean anonymous, FileOptions options) [0x00209] in /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.IO/FileStream.cs:305 
  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShare share) [0x00000] in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileStream:.ctor (string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
  at System.IO.File.OpenRead (System.String path) [0x00000] in /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.IO/File.cs:363 
  at System.IO.StreamReader..ctor (System.String path, System.Text.Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize) [0x00077] in /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.IO/StreamReader.cs:167 
  at System.IO.StreamReader..ctor (System.String path, System.Text.Encoding encoding) [0x00000] in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.IO.StreamReader:.ctor (string,System.Text.Encoding)
  at System.IO.File.ReadAllText (System.String path, System.Text.Encoding encoding) [0x00000] in /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.IO/File.cs:587 
  at System.IO.File.ReadAllText (System.String path) [0x00000] in /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.IO/File.cs:582 
  at Plugins.Editor.JetBrains.RiderAssetPostprocessor.OnGeneratedCSProjectFiles () [0x00077] in Assets/Plugins/Editor/JetBrains/Unity3DRider.cs:48 
  at Plugins.Editor.JetBrains.RiderPlugin.InitRiderPlugin () [0x00033] in Assets/Plugins/Editor/JetBrains/Unity3DRider.cs:686 
  at Plugins.Editor.JetBrains.RiderPlugin..cctor () [0x00048] in Assets/Plugins/Editor/JetBrains/Unity3DRider.cs:673

**Comment by Dante210** [comment]
If project .sln file is deleted when Unity isn't working, next time when a user tries to open Unity project he will get an exception.

**Comment by van800** [comment]
This is fixed in the plugin for Rider 2018.1 EAP and nightly builds.
Please either try it or add a manual fix to Unity3dRider.cs

Something like
```
var slnFile = RiderPlugin.SlnFile;
       if (string.IsNullOrEmpty(slnFile))
         return;

```

I would be glad to answer if you have any other questions.
